wknml/__init__.py

```diff
@@ -442,3 +450,129 @@ def write_nml(file, nml: NML):
+    for i in range(len(nodes1)):
+        if nodes1[i].id != nodes2[i].id:
+            return False
+        if nodes1[i].position != nodes2[i].position:
+            return False
+        if nodes1[i].radius != nodes2[i].radius:
+            return False
+        if nodes1[i].rotation != nodes2[i].rotation:
+            return False
+        if nodes1[i].inVp != nodes2[i].inVp:
+            return False
+        if nodes1[i].inMag != nodes2[i].inMag:
+            return False
+        if nodes1[i].bitDepth != nodes2[i].bitDepth:
+            return False
+        if nodes1[i].interpolation != nodes2[i].interpolation:
+            return False
+        if nodes1[i].time != nodes2[i].time:
+            return False
+    return True
+
+
+def compare_edges(edges1: List[Edge], edges2: List[Edge]) -> bool:
+    if len(edges1) != len(edges2):
+        return False
+    for i in range(len(edges1)):
+        if edges1[i].source != edges2[i].source:
+            return False
+        if edges1[i].target != edges2[i].target:
+            return False
+    return True
+
+
+def compare_trees(trees1: List[Tree], trees2: List[Tree]) -> bool:
+    if len(trees1) != len(trees2):
+        return False
+    for i in range(len(trees1)):
+
+        if trees1[i].id != trees2[i].id:
+            return False
+        if trees1[i].color != trees2[i].color:
+            return False
+        if trees1[i].name != trees2[i].name:
+            return False
+        if trees1[i].groupId != trees2[i].groupId:
+            return False
+        if not compare_nodes(trees1[i].nodes, trees2[i].nodes):
+            return False
+        if not compare_edges(trees1[i].edges, trees2[i].edges):
+            return False
+    return True
+
+
+def compare_branchpoints(branchpoints1: List[Branchpoint], branchpoints2: List[Branchpoint]) -> bool:
+    if len(branchpoints1) != len(branchpoints2):
+        return False
+    for i in range(len(branchpoints1)):
+        if branchpoints1[i].id != branchpoints2[i].id:
+            return False
+        if branchpoints1[i].time != branchpoints2[i].time:
+            return False
+    return True
+
+
+def compare_comments(comments1: List[Comment], comments2: List[Comment]) -> bool:
+    if len(comments1) != len(comments2):
+        return False
+    for i in range(len(comments1)):
+        if comments1[i].node != comments2[i].node:
+            return False
+        if comments1[i].content != comments2[i].content:
+            return False
+    return True
+
+
+def compare_groups(groups1: List[Group], groups2: List[Group]) -> bool:
+    if len(groups1) != len(groups2):
+        return False
+    for i in range(len(groups1)):
+        if groups1[i].id != groups2[i].id:
+            return False
+        if groups1[i].name != groups2[i].name:
+            return False
+        if not compare_groups(groups1[i].children, groups1[i].children):
+            return False
+    return True
+
+
+def compare_nml(nml1: NML, nml2: NML) -> bool:
```

Comment:
Why do we need this function and can't just do `nml1 == nml2`? Since `NML` is a named tuple, it should just work the same, I think?

Reply:
True this actually works. I thought that would not work because it did not work while I had a bug in the code that I found by implementing this method. Now we can just use the simple comparison.